web-ui/src/pages/EditSkillsPage.jsx

```diff
@@ -20,11 +20,37 @@ const EditSkillsPage = () => {
   const allSkills = selectOrderedSkills(state);
   const pendingSkills = selectPendingSkills(state);
 
+  useEffect(() => {
+    const url = new URL(location.href);
+
+    const combining = url.searchParams.get('combining');
+    setOpen(combining === 'true');
+
+    const search = url.searchParams.get('search') || '';
+    setSearchText(search);
+
+    const showAll = url.searchParams.get('showAll');
+    setShowAllSkills(showAll === 'true');
+  }, []);
+
+  useEffect(() => {
```

Comment:
Any time a setting changes, this updates the query parameters.
I used a slightly different approach this time where query parameters are only present when they have a non-default value. I should probably go back to my earlier PRs that are similar and update them to take this approach. It results in cleaner looking URLs.